Update aws-lambda-telemetry.mdx 2 (#10822)
newrelic/docs-website: src/content/docs/infrastructure/amazon-integrations/aws-integrations-list/aws-lambda-telemetry.mdx

tags:
  - Integrations
  - Amazon integrations
  - Lambda Extension
---
**Work in progress**

> The New Relic AWS Lambda Telemetry extension allows users to send telemetry data straight from their lambda functions into New Relic, with no need to set up triggers or connect Amazon CloudWatch. Send your logs, events, metrics and traces by adding the extension as the layer, and get insights from your data instently on New Relic One platform.

Reviewer: ```suggestion
The New Relic AWS Lambda Telemetry extension allows users to send telemetry data straight from their lambda functions into New Relic, with no need to set up triggers or connect Amazon CloudWatch. Send your logs, events, metrics and traces by adding the extension as the layer, and get insights from your data instantly on the New Relic platform.
```

## Installation

> 1. In the AWS Console go to **Lambda** > **Functions** and choose the function you'd like to ingest data from.

Reviewer: ```suggestion
1. In the AWS Console go to **Lambda > Functions** and choose the function you'd like to ingest data from.
```

> 2. Select **Layers** > **Add Layer**.

Reviewer: ```suggestion
2. Select **Layers > Add Layer**.
```

3. In **Choose a layer** select the **Specify a layer by providing the ARN** option.

> 4. **To Be Done** Navigate to [https://layers.newrelic-external.com/](https://layers.newrelic-external.com/), search for AwsLambdaExtension in your region to find the arn with the most recent layer version, and then paste that in the Arn field back in your AWS console. 

Reviewer: @4demos What's the **To Be Done** referring to? Should this be removed?

Author: actually now the extension is still not awilable on https://layers.newrelic-external.com/. We anticipate it to be there soon but it's not there right now.  So, now it's 4a.

> 4. **To Be Done** Navigate to [https://layers.newrelic-external.com/](https://layers.newrelic-external.com/), search for AwsLambdaExtension in your region to find the arn with the most recent layer version, and then paste that in the Arn field back in your AWS console. 

Reviewer: ```suggestion
4. **To Be Done** Navigate to [https://layers.newrelic-external.com/](https://layers.newrelic-external.com/), search for AwsLambdaExtension in your region to find the arn with the most recent layer version, and then paste that in the arn field back in your AWS console. 
```

> 4. **To Be Done** Navigate to [https://layers.newrelic-external.com/](https://layers.newrelic-external.com/), search for AwsLambdaExtension in your region to find the arn with the most recent layer version, and then paste that in the arn field back in your AWS console. 

Reviewer: ```suggestion
4. **To Be Done** Navigate to [https://layers.newrelic-external.com/](https://layers.newrelic-external.com/), search for AwsLambdaExtension in your region to find the arn with the most recent layer version, and then paste that in the `arn` field back in your AWS console. 
```

> 4a. If you can't find the extension as described in step 4, use the following pattern to get ARN <var>arn:aws:lambda:{your-region}:533243300146:layer:AwsLambdaExtension:{version}</var> The easiest way to get the most recent version would be to start from 1 and save, then push "Edit" button and select the highest version available for the chosen extension.  

Reviewer: ```suggestion
4a. If you can't find the AwsLambdaExtension, use the following pattern to get ARN:
```
 <var>arn:aws:lambda:{your-region}:533243300146:layer:AwsLambdaExtension:{version}</var>
```
To get the most recent version of the extension, start from 1 and save, click **Edit**, and select the highest version available.
```

@4demos What does "start from 1 and save" mean? Do you mean "start from **Step 1**"? Also, is arn all uppercase or lowercase? There are three variations in these steps (arn, Arn, and ARN). I can also make them codified (`arn`).

Author: no, start from version 1. arn, Arn and ARN refer to the same entity, codifying would be a good idea

Author: overall the versioning is not unified and a version can't be updated, so it's a bit tricky to get the proper (or latest) version in a given region

Reviewer: @4demos check out my updated suggestion and lmk what you think

> 5. Go to **Configuration**, select **Environment variables** and add **Key: NEW_RELIC_LICENSE_KEY  Value: <var>YOUR_LICENSE_KEY</var>**. **Key: NEW_RELIC_ACCOUNT_ID** is an optional parameter.

Reviewer: ```suggestion
5. Go to **Configuration**, select **Environment variables** and add `Key: NEW_RELIC_LICENSE_KEY  Value: <var>YOUR_LICENSE_KEY</var>`. Note: `Key: NEW_RELIC_ACCOUNT_ID` is an optional parameter.
```

> 5. Go to **Configuration**, select **Environment variables** and add **Key: NEW_RELIC_LICENSE_KEY  Value: <var>YOUR_LICENSE_KEY</var>**. **Key: NEW_RELIC_ACCOUNT_ID** is an optional parameter.

Author: could we add: "It has to be set to your Account ID on New Relic platform, if you want events data" after "NEW_RELIC_ACCOUNT_ID** is an optional parameter"?

To verify that set up has been done correctly, click the **Test** button in **code** section. 

## Find your data

> You can find your data on New Relic by navigating to **[https://one.newrelic.com/](https://one.newrelic.com/)** > Selecting one of 3 categories in the left menu: **Logs, Metrics & events, or Traces**. Traces are not shown in the default menu. To find them, go to **... Add More**, there you will find them.

Reviewer: ```suggestion
You can find your data on New Relic by navigating to **[https://one.newrelic.com/](https://one.newrelic.com/)** and  one of these categories in the left-hand menu: **Logs** or **Metrics & events**. You can also find this data on **Traces**, which is not shown in the default menu. To find it, go to **... Add More > Traces**.
```

Reviewer: @4demos Can you confirm where to find **Traces**? I'm not sure about the three dots (...) before **Add More**.

Author: "..." is what is shown in the new New Relic GUI "Add More" appears when one hovers over "..." Traces are not in the left menu by default. They need to be added if a user needs them.

Author: and it's not "this" data, it's 4th category of data.  Click on "..." gives user a tab with a bunch of other things to see so to say, Traces are there. But it's not like a sub menu when click "..." So, **...Add More > Traces** kind of simplified imho

Reviewer: @4demos Take a look at my edited suggestion to see if it makes more sense ^

tags:
  - Integrations
  - Amazon integrations
  - Lambda Extension
---
**Work in progress**

> The New Relic AWS Lambda Telemetry extension allows users to send telemetry data straight from their lambda functions into New Relic, with no need to set up triggers or connect Amazon CloudWatch. Send your logs, events, metrics and traces by adding the extension as the layer, and get insights from your data instantly on the New Relic platform.

Reviewer: ```suggestion
The New Relic AWS Lambda Telemetry extension allows users to send telemetry data straight from their lambda functions into New Relic, with no need to set up triggers or connect Amazon CloudWatch. Send your logs, events, metrics, and traces by adding the extension as the layer, and get insights from your data instantly on the New Relic platform.
```

## Installation

1. In the AWS Console go to **Lambda > Functions** and choose the function you'd like to ingest data from.

> 2. Select **Layers** > **Add Layer**.

Reviewer: ```suggestion
2. Select **Layers > Add Layer**.
```
